### PR TITLE
[PW-6427] - GraphQL add cart id

### PR DESCRIPTION
**Pull request by Morerice**
**Description**
This PR will add the `masked_cart_id` requirement to the **GetAdyenPaymentStatus** and the **GetAdyenPaymentDetails** resolvers.

Hence, the **GetAdyenPaymentStatus** will have to be made as a specific separate call, instead of automatically included in the placeOrder resolver, since this resolver does not return the cart_id automatically.

**Tested scenarios**
* Adyen payment methods
* Adyen payment status - guest
* Adyen payment status - logged in
* Adyen payment details - invalid payload
* Adyen payment details - incorrect order number
* Adyen payment details - correct

### CODE REVIEW

**Comment by acampos1916** [comment]
Hey @larsroettig would you happen to have a few minutes to look at this PR? We're interested on what you think having worked with the GraphQL functionality of this plugin. I mentioned that we wanted to update some queries. Appreciate it!

**Comment by larsroettig** [comment]
@acampos1916 many services like `DataProvider\GetAdyenPaymentStatus` throw `LocalizedException` but in production mode you get only the information internal server error  this makes it impossible to evaluate them in the frontend or build proper handling for differnt cases  an messages for the customer

**Comment by larsroettig** [comment]
@acampos1916 many services like `DataProvider\GetAdyenPaymentStatus` throw `LocalizedException` but in production mode you get only the information internal server error  this makes it impossible to evaluate them in the frontend or build proper handling for differnt cases  an messages for the customer

File: Helper/Quote.php [hunk 1]
```diff
@@ -130,4 +139,60 @@ public function getIsQuoteMultiShippingWithMerchantReference(string $merchantRef
...
+        try {
+            /** @var QuoteModel $cart */
+            $cart = $this->cartRepository->get($cartId);
+        } catch (NoSuchEntityException $e) {
+            throw new NoSuchEntityException(
+                __('Could not find a cart with ID "%masked_cart_id"', ['masked_cart_id' => $cartHash])
+            );
+        }
+
+        if ($cart->getIsActive()) {
```
Comment (AlexandrosMor) [inline Helper/Quote.php:171]: Why do we throw an exception if a cart is active?

Comment (Morerice) [reply Helper/Quote.php:171]: Because it shouldn't be active since these calls will be made after the order has been completed

Comment (AlexandrosMor) [reply Helper/Quote.php:171]: ok got it

**Review by AlexandrosMor — APPROVED** [review]

**Comment by Morerice** [comment]
> @acampos1916 many services like `DataProvider\GetAdyenPaymentStatus` throw `LocalizedException` but in production mode you get only the information internal server error this makes it impossible to evaluate them in the frontend or build proper handling for differnt cases an messages for the customer

Hi @larsroettig,

Thanks for your input. This is a very valid point, but ultimately it's not something that was introduced in this PR or in GraphQL. Along with the stateData issue, this is also something that we want to look into.

Regards,
Jean
Adyen

**Comment by larsroettig** [comment]
I see two problems here first, I have to create a new shopping cart before I can forward the customer to 3D Secure 2 or 3D Secure 1 otherwise, the customer will be logged out in PWAStudio for security reasons. Currently, there is no excellent way to get around this spend two days in the investigation where I need to have a valid cart token. 


 Secondly, When you remove `adyen_payment_status` from the place order call, this would mean you need an additional server request. I here no added value for the community would even have to implement the old end point because there is no way in PWAStudio.

**Comment by Morerice** [comment]
Hi @larsroettig 

I updated the endpoint using the interceptor as discussed in our meeting. Could you please verify that this is also functional for you? Example request:
```
mutation setPaymentMethod(
    $cartId: String!
) {
    setPaymentMethodOnCart(
        input: {
            cart_id: $cartId
            payment_method: {
                code: "adyen_cc"
                adyen_additional_data_cc: {
                    cc_type: "VI"
                    stateData:  ""
                }
            }    
        }
    ) {
        cart {
            selected_payment_method {
                code
                title
            }
        }
    }

    placeOrder(
        input: {
            cart_id: $cartId
        }
    ) {
        order {
            order_number
            cart_id
            adyen_payment_status {
                isFinal
                resultCode
                additionalData
                action
            }
        }
    }
}
```

Best regards,
Jean
Adyen

**Comment by larsroettig** [comment]
setPaymentMethodOnCart and placeOrder in same call mean you have to save dropin infermation in a context to make working with PWA Studio thought you wanted to change that too. But will try store this information in client RAM via a context

**Comment by larsroettig** [comment]
@Morerice thanks have created ticket and start working on testing hope by end tomorrow i can say something

**Review by peterojo — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Comment by larsroettig** [comment]
@Morerice I remove our plugin combine in one request 
```mutation setPaymentMethod(
    $cartId: String!
) {
    setPaymentMethodOnCart(
        input: {
            cart_id: $cartId
            payment_method: {
                code: "adyen_cc"
                adyen_additional_data_cc: {
                    cc_type: "VI"
                    stateData:  ""
                }
            }    
        }
    ) {
        cart {
            selected_payment_method {
                code
                title
            }
        }
    }

    placeOrder(
        input: {
            cart_id: $cartId
        }
    ) {
        order {
            order_number
            cart_id
            adyen_payment_status {
                isFinal
                resultCode
                additionalData
                action
            }
        }
    }
}```

**Pull request merged**